com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateLoop.cs

```diff
@@ -0,0 +1,525 @@
+            {
+                case NetworkUpdateStage.Initialization:
+                {
+                    if (m_Initialization_List.Contains(updateSystem))
+                    {
+                        m_Initialization_List.Remove(updateSystem);
+                        m_Initialization_Array = m_Initialization_List.ToArray();
+                    }
+
+                    break;
+                }
+                case NetworkUpdateStage.EarlyUpdate:
+                {
+                    if (m_EarlyUpdate_List.Contains(updateSystem))
+                    {
+                        m_EarlyUpdate_List.Remove(updateSystem);
+                        m_EarlyUpdate_Array = m_EarlyUpdate_List.ToArray();
+                    }
+
+                    break;
+                }
+                case NetworkUpdateStage.FixedUpdate:
+                {
+                    if (m_FixedUpdate_List.Contains(updateSystem))
+                    {
+                        m_FixedUpdate_List.Remove(updateSystem);
+                        m_FixedUpdate_Array = m_FixedUpdate_List.ToArray();
+                    }
+
+                    break;
+                }
+                case NetworkUpdateStage.PreUpdate:
+                {
+                    if (m_PreUpdate_List.Contains(updateSystem))
+                    {
+                        m_PreUpdate_List.Remove(updateSystem);
+                        m_PreUpdate_Array = m_PreUpdate_List.ToArray();
+                    }
+
+                    break;
+                }
+                case NetworkUpdateStage.Update:
+                {
+                    if (m_Update_List.Contains(updateSystem))
+                    {
+                        m_Update_List.Remove(updateSystem);
+                        m_Update_Array = m_Update_List.ToArray();
+                    }
+
+                    break;
+                }
+                case NetworkUpdateStage.PreLateUpdate:
+                {
+                    if (m_PreLateUpdate_List.Contains(updateSystem))
+                    {
+                        m_PreLateUpdate_List.Remove(updateSystem);
+                        m_PreLateUpdate_Array = m_PreLateUpdate_List.ToArray();
+                    }
+
+                    break;
+                }
+                case NetworkUpdateStage.PostLateUpdate:
+                {
+                    if (m_PostLateUpdate_List.Contains(updateSystem))
+                    {
+                        m_PostLateUpdate_List.Remove(updateSystem);
+                        m_PostLateUpdate_Array = m_PostLateUpdate_List.ToArray();
+                    }
+
+                    break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// The current network update stage being executed.
+        /// </summary>
+        public static NetworkUpdateStage UpdateStage;
+
+        private static readonly List<INetworkUpdateSystem> m_Initialization_List = new List<INetworkUpdateSystem>();
```

Comment:
aaand I forgot to attach `List` vs `HashSet` performance comparison to the PR description — I will do that soon :)

Reply:
`HashSet` turned out to be the better choice — `HashSet`'s `Add` and `Remove` are faster than `List`'s. @mattwalsh-unity FYI :)